### PR TITLE
Add ENV.size, ENV.to_h, ENV.delete, ENV.has_key?, ENV.assoc...

### PR DESCRIPTION
**Pull request by nanobowers**
Add `ENV.size`, `ENV.to_h`, `ENV.delete`, `ENV.has_key?`, `ENV.assoc`, `ENV.fetch`, `ENV.to_s` and `ENV.rehash` and related aliased functions (e.g. `ENV.include?`, `ENV.member`, `ENV.store`, etc).

This PR should enable 12 added spec files to pass all expectations.  I tried to implement `ENV.clear` and `ENV.replace` (which arguably should rely on `ENV.clear`) but had issue with crashes on subsequent ENV accesses post-clear and then removed/abandoned those two methods.  Getting those working will be very useful in getting more of the `ENV` test suite passing since they're used extensively.

### CODE REVIEW

File: src/env_object.cpp [hunk 1]
```diff
@@ -7,9 +7,14 @@ extern char **environ;
 namespace Natalie {
 
 Value EnvObject::inspect(Env *env) {
+    return this->to_hash(env)->as_hash()->inspect(env);
+}
+
+Value EnvObject::to_hash(Env *env) {
```
Comment (nanobowers) [inline src/env_object.cpp:13]: Refactored `::to_hash` out of `::inspect` to use it as a standalone function

File: src/env_object.cpp [hunk 2]
```diff
@@ -19,24 +24,89 @@ Value EnvObject::inspect(Env *env) {
...
     char *value = getenv(name->as_string()->c_str());
     if (value) {
         return new StringObject { value };
     } else {
-        return NilObject::the();
+        env->raise("KeyError", "key not found: {}", name->as_string()->string());
     }
 }
 
 Value EnvObject::refeq(Env *env, Value name, Value value) {
```
Comment (nanobowers) [inline src/env_object.cpp:84]: `EnvObject::ref` and `EnvObject::refeq` have been fixed to handle corner cases per the spec tests.

File: lib/natalie/compiler/binding_gen.rb [hunk 3]
```diff
@@ -468,9 +468,24 @@ def generate_name
...
+gen.singleton_binding('ENV', 'has_key?', 'EnvObject', 'has_key', argc: 1, pass_env: true, pass_block: false, return_type: :bool)
+gen.singleton_binding('ENV', 'include?', 'EnvObject', 'has_key', argc: 1, pass_env: true, pass_block: false, return_type: :bool)
+gen.singleton_binding('ENV', 'member?', 'EnvObject', 'has_key', argc: 1, pass_env: true, pass_block: false, return_type: :bool)
+gen.singleton_binding('ENV', 'to_h', 'EnvObject', 'to_hash', argc: 0, pass_env: true, pass_block: false, return_type: :Object)
+gen.singleton_binding('ENV', 'to_hash', 'EnvObject', 'to_hash', argc: 0, pass_env: true, pass_block: false, return_type: :Object)
+gen.singleton_binding('ENV', 'size', 'EnvObject', 'size', argc: 0, pass_env: true, pass_block: false, return_type: :Object)
+gen.singleton_binding('ENV', 'length', 'EnvObject', 'size', argc: 0, pass_env: true, pass_block: false, return_type: :Object)
+gen.singleton_binding('ENV', 'to_s', 'EnvObject', 'to_s', argc: 0, pass_env: false, pass_block: false, return_type: :Object)
+gen.singleton_binding('ENV', 'rehash', 'EnvObject', 'rehash', argc: 0, pass_env: false, pass_block: false, return_type: :Object)
```
Comment (seven1m) [inline lib/natalie/compiler/binding_gen.rb:488]: Let’s put this list of methods in alphabetical order.

**Review by seven1m — APPROVED** [review]
Looks good!

File: lib/natalie/compiler/binding_gen.rb (hunk 3 above)
Comment (nanobowers) [reply lib/natalie/compiler/binding_gen.rb]: Done with latest push.

Comment (seven1m) [reply lib/natalie/compiler/binding_gen.rb]: Thanks!
